src/triangle.rs

```diff
@@ -128,7 +128,11 @@ impl Triangle {
         let bc = self.b.distance(self.c);
         let ac = self.a.distance(self.c);
 
-        Circle::new(self.circumcenter(), 2.0 * self.area() / (ab + bc + ac))
+        let perimeter_recip = 1. / (ab + bc + ac);
+        let incenter = (self.a.to_vec2() * bc + self.b.to_vec2() * ac + self.c.to_vec2() * ab)
+            * perimeter_recip;
```

Comment:
It might be cleaner to just divide by (ab + bc + ac) here, as the Div implementation already does the recip optimization.

Reply:
Ah, indeed! I remember seeing that, but had forgotten it in the meantime.

In this specific case, line 135 calculates `2.0 * self.area() * perimeter_recip`. The compiler is smart enough to still only do the recip calculation once when dividing the Vec2 by `(ab + bc + ac)`, but maybe it's clearer this way anyway that there's only one division instruction.

Reply:
Added a code comment.